src/tasks/submit.rs

```diff
@@ -48,6 +48,133 @@ macro_rules! spawn_provider_send {
     };
 }
 
+/// Represents the kind of revert that can occur during simulation.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum SimRevertKind {
+    /// Incorrect host block error
+    IncorrectHostBlock,
+    /// Bad signature error
+    BadSignature,
```

Comment:
I assume we're not preserving the data in tihs error since it's probably not worth it do so anyway

Reply:
it's an error _kind_ not an error itself. it's just a memoization of the type of the actual error data stored elsewhere